pykokkos/core/parsers/parser.py

```diff
@@ -123,19 +123,186 @@ def get_entities(self, style: PyKokkosStyles) -> Dict[str, PyKokkosEntity]:
 
         for i, node in enumerate(self.tree.body):
             if check_entity(node, self.pk_import):
-                start: int = node.lineno - 1
 
+                start: int = node.lineno - 1
                 try:
                     stop: int = self.tree.body[i + 1].lineno - 1
                 except IndexError:
                     stop = len(self.lines)
 
                 name: str = node.name
+
                 entity = PyKokkosEntity(style, cppast.DeclRefExpr(name), node, (self.lines[start:stop], start), self.path, self.pk_import)
                 entities[name] = entity
 
         return entities
 
+
+    def fix_types(self, entity: PyKokkosEntity, updated_types: UpdatedTypes) -> ast.AST:
+        '''
+        Inject (into the entity AST,) the missing annotations for datatypes that have been inferred 
+
+        entity: Pykokkos entity whose AST will be patched - the entity being compiled/translated
+        updated_types: UpdatedTypes object that contains info about inferred types for this entity
+        returns: the updated entity AST after injecting correct annotations (from updated_types) for datatypes
+        '''
+
+        style: PyKokkosStyles = entity.style
+        assert style is PyKokkosStyles.workunit and updated_types is not None
+
+        # For now, just so we can raise an error instead of unexpectedly crashing
+        primitives_supported = ["int", "bool", "float"]
+
+        entity_tree: ast.AST = entity.AST
+
+        # if modifications to layout decorator is needed
+        if len(updated_types.layout_change):
+            entity_tree.decorator_list = self.fix_view_layout(entity_tree, updated_types.layout_change)
+
+        for arg_obj in entity_tree.args.args:
+            # Type already provided by the user
+            if arg_obj.arg not in updated_types.inferred_types:
```

Comment:
you could check here if the user type matches inferred type

Reply:
We don't infer any type the user already provided for. We decided this to let user "force" a type in cases were inference yields unwanted results